lib/Post.php

```diff
@@ -1498,53 +1498,148 @@ public function paged_content() {
 	}
 
 	/**
-	 * Get the date to use in your template!
+	 * Gets the timestamp when the post was published.
+	 *
+	 * @api
+	 * @since 2.0.0
+	 *
+	 * @return false|int Unix timestamp on success, false on failure.
+	 */
+	public function timestamp() {
+		return get_post_timestamp( $this->ID );
+	}
+
+	/**
+	 * Gets the timestamp when the post was last modified.
+	 *
+	 * @api
+	 * @since 2.0.0
+	 *
+	 * @return false|int Unix timestamp on success, false on failure.
+	 */
+	public function modified_timestamp() {
+		return get_post_timestamp( $this->ID, 'modified' );
+	}
+
+	/**
+	 * Gets the publishing date of the post to use in your template.
 	 *
 	 * @api
 	 * @example
 	 * ```twig
-	 * Published on {{ post.date }} // Uses WP's formatting set in Admin
+	 * {# Uses date format set in Settings → General #}
+	 * Published on {{ post.date }}
 	 * OR
-	 * Published on {{ post.date('F jS') }} // Jan 12th
+	 * Published on {{ post.date('F jS') }}
 	 * ```
 	 *
 	 * ```html
 	 * Published on January 12, 2015
 	 * OR
 	 * Published on Jan 12th
 	 * ```
-	 * @param string $date_format
+	 *
+	 * @param string|null $date_format Optional. PHP date format. Will use the `date_format` option
+	 *                                 as a default.
+	 *
 	 * @return string
 	 */
-	public function date( $date_format = '' ) {
-		$df       = $date_format ? $date_format : get_option('date_format');
-		$the_date = date_i18n($df, strtotime($this->post_date));
-		return apply_filters('get_the_date', $the_date, $df);
+	public function date( $date_format = null ) {
+		$format = $date_format ? $date_format : get_option( 'date_format' );
```

Comment:
This and other instances of this pattern can all be updated to the shorthand ternary operator, `?:`:

```
$format = $date_format ?: get_option( 'date_format' );
```

Reply:
I like your suggestion to use the more modern shorthand syntaxes. Updated in https://github.com/timber/timber/pull/2137/commits/1c84ffc1d313a686b9cbd8d1c0a73604e6ea1720.